data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/AbstractBeaconHandlerTest.java

```diff
@@ -57,6 +63,14 @@ protected void verifyStatusCode(final int statusCode) {
     return jsonProvider.jsonToObject(val, clazz);
   }
 
+  protected <T> T getResponseFromFuture(Class<T> clazz)
+      throws ExecutionException, InterruptedException, JsonProcessingException {
+    verify(context).result(args.capture());
+    SafeFuture<String> future = args.getValue();
+    String data = future.get();
```

Comment:
Wes should check the future is actually completed first so we don't hang forever:
```suggestion
    assertThat(future).isCompleted();
    String data = future.join();
```

Then using `join` instead of `get` will let you drop most of the declared exceptions.